sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1648,16 +1648,56 @@ object RewriteCorrelatedScalarSubquery extends Rule[LogicalPlan] {
   }
 
   /**
+   * Statically evaluate an expression containing one or more aggregates on an empty input.
+   */
+  private def evalOnZeroTups(expr : Expression) : Option[Any] = {
+    // AggregateExpressions are Unevaluable, so we need to replace all aggregates
+    // in the expression with the value they would return for zero input tuples.
+    val rewrittenExpr = expr transform {
+      case a @ AggregateExpression(aggFunc, _, _, resultId) =>
+        val resultLit = aggFunc.defaultResult match {
+          case Some(lit) => lit
+          case None => Literal.default(NullType)
+        }
+        Alias(resultLit, "aggVal") (exprId = resultId)
+    }
+    Option(rewrittenExpr.eval())
+  }
+
+  /**
    * Construct a new child plan by left joining the given subqueries to a base plan.
    */
   private def constructLeftJoins(
       child: LogicalPlan,
       subqueries: ArrayBuffer[ScalarSubquery]): LogicalPlan = {
     subqueries.foldLeft(child) {
       case (currentChild, ScalarSubquery(query, conditions, _)) =>
+        val aggOutputExpr = query.asInstanceOf[Aggregate].aggregateExpressions.head
```

Comment:
Interesting. I had thought that the checks in the Analyzer would ensure that only Aggregate nodes could appear immediately under a ScalarSubquery node. Obviously that is not the case.